docs/proposals/25-private-cloud-profiles.md

```diff
@@ -0,0 +1,299 @@
+
+This GEP proposes a mechanism that will allow project administrators to create private cloud profiles that are only visible to and can only be consumed by the project they were created in.
+
+## Motivation
+
+### Context
+
+Cloud profiles are an integral component of Gardener for managing shoot resources. They are currently managed centrally by Gardener and can be consumed by any Gardener user. However, some teams require frequent changes to a cloud profile. There are a few reasons why:
+
+1. Testing with different machine types than the ones present in the cloud profile.
+2. Need to use different volume types than the ones present in the cloud profile.
+3. Extending the expiration date of Kubernetes versions. Teams might need to hang on to an older version of Kubernetes as they might not be ready for an upgrade as of the set expiration date. Given that the cloud profile is a cluster-scoped resource, it is currently not possible to extend the expiration date for shoots in one project but only centrally for all projects.
+4. Extending the expiration date for machine images. For the same reasons as extending the expiration date of the Kubernetes versions.
+5. Some gardener users might have access to custom cloud provider regions that others do not have access to, they currently have no way of using custom cloud provider regions.
+
+### Current State
+
+Cloud Profiles are non-namespaced resources. This means that in a typical gardener installation, only a handful of cloud profiles (i.e. one per cloud provider) exist. They can be consumed by any project's clusters. Consequently, when a project requires changes to cloud profiles for any of the reasons mentioned in the [context section](#context), they are changed for the entire gardener landscape. Since some projects might require frequent changes to a cloud profile, this become quite a cumbersome process. Additionally, it adds confusion to every project's workflow if say for example a team requires an extension of the expiration date of a Kubernetes version since they will be wondering why it was extended.
+
+### Goals
+
+- Reduce load on Gardener operators for maintaining cloud profiles
+- Possibly reduce the time a team has to wait until a change in a cloud profile is reflected in production
+- Make it so that project-scoped information in the cloud profile is only visible to the relevant project
+- Full backward compatibility
+
+### Non-Goals
+
+- Automate approval process for changes in Gardeners cloud profiles
+- Add different, possibly unsupported Kubernetes versions or machine image versions and names
+
+## Proposal
+
+It is proposed to implement a solution that enables project administrators to create custom (private) cloud profiles. These cloud profiles would be consumed by project users of the project they were created in.
+
+### Approach
```

Comment:
After re-iterating over `CloudProfile`s of known providers, I think leaving out the `providerConfig` will restrict this feature massively but even worse, `PrivateCloudProfile`s create an impression that a configuration of fields like `regions` and `machineImages` just work, but most of the time additional configuration is required.

Examples:
1. To enable a new region in AWS, it's required to configure AMI(s) in `providerConfig`.
2. To enable a new region in OpenStack, floating pools and images must be maintained in `providerConfig`.
3. To extend an expiration date of an OS in GCP, Gardener Operators must not remove the image in `providerConfig`. 

I think we should evaluate supporting `providerConfig`:
- By default, the PrivateCloudProfile-Controller will just consider the parent `providerConfig` when it writes the merged profile to `.status`.
- Provider extensions can optionally register a webhook that is capable of merging the `providerConfig` portion  as intended.

Any opinions @rfranzke?

Reply:
I am not against it, just voted to not look at it right now to avoid the additional complexity to build this webhook for all provider extensions. Feel free to include and implement this if you want to enable such use cases.